controller/resource_roles.go

```diff
@@ -33,15 +30,28 @@ func (c *ResourceRolesController) ListAssigned(ctx *app.ListAssignedResourceRole
 
 	var roles []role.IdentityRole
 
-	roles, err := c.roleManagementService.ListByResource(ctx, ctx.ResourceID)
+	err := application.Transactional(c.db, func(appl application.Application) error {
```

Comment:
This is where the new design shows up - basically the controller calling the 'model' and repository.